pkg/preflight/read_specs_test.go

```diff
@@ -0,0 +1,299 @@
+package preflight
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/replicatedhq/troubleshoot/internal/testutils"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PreflightSpecsReadTests []PreflightSpecsReadTest
+
+type PreflightSpecsReadTest struct {
+	name          string
+	args          []string
+	customStdin   bool
+	stdinDataFile string
+	//
+	wantErr           bool
+	wantPreflightSpec *troubleshootv1beta2.Preflight
+	// TODOLATER: tests around this
+	wantHostPreflightSpec *troubleshootv1beta2.HostPreflight
+	// TODOLATER: tests around this
+	wantUploadResultSpecs []*troubleshootv1beta2.Preflight
+}
+
+func TestPreflightSpecsRead(t *testing.T) {
+	// NOTE: don't use t.Parallel(), these tests manipulate os.Stdin
+
+	// A very simple preflight spec
+	preflightFile := filepath.Join(testutils.FileDir(), "../../testdata/preflightspec/troubleshoot_v1beta2_preflight_gotest.yaml")
+	//
+	expectPreflightSpec := troubleshootv1beta2.Preflight{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Preflight",
+			APIVersion: "troubleshoot.sh/v1beta2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "go-test-preflight",
+		},
+		Spec: troubleshootv1beta2.PreflightSpec{
+			UploadResultsTo: "",
+			Collectors: []*troubleshootv1beta2.Collect{
+				&troubleshootv1beta2.Collect{
```

Comment:
just a note. it's not necessary to manually type these structs since you've already set the type for the slice.